engine/settings/base.py

```diff
@@ -84,21 +81,94 @@
 # Outgoing webhook settings
 DANGEROUS_WEBHOOKS_ENABLED = getenv_boolean("DANGEROUS_WEBHOOKS_ENABLED", default=False)
 
-# DB backend defaults
-DB_BACKEND = os.environ.get("DB_BACKEND", "mysql")
-DB_BACKEND_DEFAULT_VALUES = {
+# Database
+DATABASE_DEFAULTS = {
     "mysql": {
         "USER": "root",
-        "PORT": "3306",
+        "PORT": 3306,
+    },
+    "postgresql": {
+        "USER": "postgres",
+        "PORT": 5432,
+    },
+}
+
+DATABASE_NAME = os.getenv("DATABASE_NAME") or os.getenv("MYSQL_DB_NAME")
+DATABASE_USER = os.getenv("DATABASE_USER") or os.getenv("MYSQL_USER")
+DATABASE_PASSWORD = os.getenv("DATABASE_PASSWORD") or os.getenv("MYSQL_PASSWORD")
+DATABASE_HOST = os.getenv("DATABASE_HOST") or os.getenv("MYSQL_HOST")
+DATABASE_PORT = os.getenv("DATABASE_PORT") or os.getenv("MYSQL_PORT")
+
+DATABASE_TYPE = os.getenv("DATABASE_TYPE", "mysql").lower()
+assert DATABASE_TYPE in ["mysql", "postgresql", "sqlite3"]
```

Comment:
Probably not a big difference in this case, but this check would be more efficient using a set